connect-utils/src/main/java/com/github/jcustenborder/kafka/connect/utils/RegexUtils.java

```diff
@@ -0,0 +1,265 @@
+      try {
+        ForkJoinPool.managedBlock(executor);
+        return executor.getStringResult();
+      } catch (InterruptedException e) {
+        Thread.currentThread().interrupt();
+        throw new CompletionException(e);
+      }
+    });
+
+    try {
+      return future.get(timeoutMs, TimeUnit.MILLISECONDS);
+    } catch (TimeoutException e) {
+      log.warn(
+          "Regex operation exceeded timeout of {} ms. Returning original string.",
+          timeoutMs, e);
+      // Do not use the result of the regex operation after a timeout
+      return null; // Signal to caller that a timeout occurred
+    }
+  }
+
+  private static boolean executeBooleanRegexOperation(
+      Pattern pattern,
+      String input,
+      OperationType operationType,
+      long timeoutMs) throws InterruptedException, ExecutionException {
+
+    if (input == null) {
+      return false;
+    }
+
+    RegexExecutor executor = new RegexExecutor(pattern, input, null, operationType);
+    CompletableFuture<Boolean> future = CompletableFuture.supplyAsync(() -> {
+      try {
+        ForkJoinPool.managedBlock(executor);
+        return executor.getBooleanResult();
+      } catch (InterruptedException e) {
+        Thread.currentThread().interrupt();
+        throw new CompletionException(e);
+      }
+    });
+
+    try {
+      return future.get(timeoutMs, TimeUnit.MILLISECONDS);
+    } catch (TimeoutException e) {
+      log.warn(
+          "Regex operation exceeded timeout of {} ms. Returning false.",
+          timeoutMs, e);
+      // On timeout, return false
+      return false;
+    }
+  }
+
+  /**
+   * Executes a regex replacement with timeout protection.
+   * See class-level JavaDoc for details on ReDoS protection.
+   *
+   * @param input        The input string
+   * @param replacements The regex operation to perform
+   * @param timeoutMs    The timeout in milliseconds
+   * @return The result of the operation
+   * @throws InterruptedException if the current thread is interrupted
+   * @throws ExecutionException   if the operation throws an exception
+   */
+  public static String replaceAll(
+      String input,
+      Map<Pattern, String> replacements,
+      long timeoutMs) throws InterruptedException, ExecutionException {
+
+    if (input == null) {
+      return null;
+    }
+    if (replacements == null || replacements.isEmpty()) {
+      return input;
+    }
+
+    String currentResult = input;
+    for (Map.Entry<Pattern, String> entry : replacements.entrySet()) {
+      String result = executeStringRegexOperation(
+          entry.getKey(),
+          currentResult,
+          entry.getValue(),
+          timeoutMs
+      );
+      if (result == null) {
+        // Timeout occurred, return original input
+        return input;
+      }
+      currentResult = result;
+    }
+    return currentResult;
+  }
+
+  /**
+   * Executes a regex find operation with timeout protection.
```

Comment:
nit: similarly for other javadocs
```suggestion
   * Executes a {@link Pattern#matcher(CharSequence)#find(Pattern, String, long)}  } operation with timeout protection.
```

Reply:
done